### ISSUE TITLE
Define a strategy for certs for brokers

### ISSUE THREAD

**Issue by vaikas-google** [kubernetes-incubator/service-catalog#606]
Some brokers (especially internal ones, or testing ones) do not have proper certs and we probably do not even install proper roots in our containers. This issue tracks defining a procedure for doing this:
 1. install proper root certs
 2. define a process for adding additional ones for organizations

**Comment by arschles** [comment]
@vaikas-google do you consider the following to be a solution:

1. Install standard CA roots to the [controller-manager `Dockerfile`](https://github.com/kubernetes-incubator/service-catalog/blob/master/build/controller-manager/Dockerfile)
2. Default to standard TLS (without `InsecureSkipVerify`) for all controller-to-broker communication
3. Add a new `brokerSecurity` field to the `Broker` resource, which has the following possible values: `full` (default), `skipVerify`, and `none`

**Comment by arschles** [comment]
Punting this to beta to decide.

**Comment by vaikas-google** [comment]
Yeah, so I think that this works for certs from roots. I'd be curious if there are organizations that would be running with their own CAs and in that case how we would be adding additional certs for those.

**Comment by pmorie** [comment]
Idea: control insecure-skip-verify like so:

```go
type BrokerSpec struct {
  // other fields omitted

  Insecure *bool
}
```

**Comment by arschles** [comment]
Moving to 1.0.0, since we can already support brokers with root CAs now

**Comment by pmorie** [comment]
I have a strong need for this in beta.  Adding to the agenda to discuss in the July 24 2017 SIG meeting.

**Comment by mrbobbytables** [comment]
This project is being archived, closing open issues and PRs. 
Please see this PR for more information: https://github.com/kubernetes/community/pull/6632
